### DISCUSSION TITLE
Solis S5-EH1P  compatibily

### DISCUSSION THREAD

**Discussion by kiloptero** [wills106/homeassistant-solax-modbus#679] [General]
hi all! anyone here with a S5-EH1P ? im waiting to gain more info to purchase this.. using the waveshare looks easy , but dont know about the compatibility! thanks a lot!!

**Comment by thekeymeister** [comment]
I've been successfully using a Solis S5-EH1P6K-L with this integration for around 6 months now, in conjunction with a Waveshare.

**Comment by kiloptero** [comment]
waveshare to use both ( cloud and local)?

Andres Roepke Del Solar.
Ingeniero Civil Industrial
Ingeniero Electronico
Mobile 88196932

**Comment by thekeymeister** [comment]
I did it almost exactly like this, two Waveshares (I am using lead acid batteries so no need for a third Waveshare for BMS comms in my case).  

https://github.com/alienatedsec/solis-ha-modbus-cloud

Both SolisCloud and HA are working fine together, I very rarely use SolisCloud any more but it took a while to get HA working as I wanted it, so it was definitely beneficial to have both available.

**Comment by kiloptero** [comment]
Can you share the way yo use both?

El jue, 4 ene 2024 a las 14:11, Andres Roepke ***@***.***>) escribió:

> waveshare to use both ( cloud and local)?
>
> Andres Roepke Del Solar.
> Ingeniero Civil Industrial
> Ingeniero Electronico
> Mobile 88196932
>
>
> On Thu, Jan 4, 2024 at 14:05 thekeymeister ***@***.***>
> wrote:
>
>> I've been successfully using a Solis S5-EH1P6K-L with this integration
>> for around 6 months now, in conjunction with a Waveshare.
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/wills106/homeassistant-solax-modbus/discussions/679#discussioncomment-8015134>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AIOTKWRA4FG6RFNPWDL4ROLYM3ONJAVCNFSM6AAAAABBDT4PQKVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DAMJVGEZTI>
>> .
>> You are receiving this because you authored the thread.Message ID:
>> <wills106/homeassistant-solax-modbus/repo-discussions/679/comments/8015134
>> @github.com>
>>
>

-- 
Andres Roepke Del Solar.
Ingeniero Civil Industrial
Ingeniero Electronico
Mobile 88196932

**Comment by kiloptero** [comment]
Thanks.. Can you share your dashboard or the final stage of your journey?
maybe i can copy and take good ideas!!

El jue, 4 ene 2024 a las 14:18, Andres Roepke ***@***.***>) escribió:

> Can you share the way yo use both?
>
> El jue, 4 ene 2024 a las 14:11, Andres Roepke ***@***.***>)
> escribió:
>
>> waveshare to use both ( cloud and local)?
>>
>> Andres Roepke Del Solar.
>> Ingeniero Civil Industrial
>> Ingeniero Electronico
>> Mobile 88196932
>>
>>
>> On Thu, Jan 4, 2024 at 14:05 thekeymeister ***@***.***>
>> wrote:
>>
>>> I've been successfully using a Solis S5-EH1P6K-L with this integration
>>> for around 6 months now, in conjunction with a Waveshare.
>>>
>>> —
>>> Reply to this email directly, view it on GitHub
>>> <https://github.com/wills106/homeassistant-solax-modbus/discussions/679#discussioncomment-8015134>,
>>> or unsubscribe
>>> <https://github.com/notifications/unsubscribe-auth/AIOTKWRA4FG6RFNPWDL4ROLYM3ONJAVCNFSM6AAAAABBDT4PQKVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DAMJVGEZTI>
>>> .
>>> You are receiving this because you authored the thread.Message ID:
>>> <wills106/homeassistant-solax-modbus/repo-discussions/679/comments/8015134
>>> @github.com>
>>>
>>
>
> --
> Andres Roepke Del Solar.
> Ingeniero Civil Industrial
> Ingeniero Electronico
> Mobile 88196932
>


-- 
Andres Roepke Del Solar.
Ingeniero Civil Industrial
Ingeniero Electronico
Mobile 88196932

**Comment by thekeymeister** [comment]
See attached a screenshot of my setup,
I am using the backup/EPS port on my inverter to supply essential loads (lighting, sockets etc) - so the load view is split between essential and non essential.  "Tower of Power" is what I lovingly call my lead acid battery bank.

I have used Solcast to do my solar forecasting as it was easier and more accurate than Forecast.Solar.
I am currently running an automation to decide whether to charge my batteries from the grid at the cheapest grid period, this takes a two day average of the house load and compares it against the forecast solar production for the following day, and if the battery voltage is below a certain threshold it will set an input boolean which is then used in a separate automation triggered at the cheapest grid period.

I tried for a while to set the charging periods to follow the cheapest grid periods, but it became very complicated.  Instead I have set the charging period to 00:00-23:59 and turn charging on by changing the 'Solax Energy Storage Control Switch' to either 'Timed Charge/Discharge' or 'Selfuse - No grid charging' when triggered by relevant automations I have running.

My other tabs (sensors / controls) just contain all of the sensors and switches, but they are not used by me very often.

Hope this helps!

![Solar](https://github.com/wills106/homeassistant-solax-modbus/assets/142272307/a53ad4a4-33c4-4218-9b55-25065f755d8e)

**Comment by kiloptero** [comment]
This is great, thank you so much!
thanks amazing!!
i
will create an automation ( in the future) to start charging in a specific
hour ( we have 3 dif price time period) and discharg in the night for start

then maybe when the battery charge more than the consume of
the house .. try and error :)

**Comment by chronvas** [comment]
I have just finished connecting my Solis S5-EH1P3K-L using this method https://github.com/Jumpy07/Solis---SolisCloud-and-Home-Assistant (no complicated waveshare stuff) and I've kept my SolisCloud intact.
The only thing I had to change is the DEFAULT_BAUDRATE to 9600 in the plugin `/config/custom_components/solax_modbus/const.py`.

I am not sure if this is working 100% as expected, I see 145 Entities and I will dig into them some time in the future. The state of charge which I was interested reeports fine but only every 5 minutes, even if I have my setup to update every 10 seconds. I will have to see how that goes.
